### ISSUE TITLE
Skins not loading in profiles

### ISSUE THREAD

**Issue by Lanternon** [xbmc/xbmc#14732]
### Describe the bug
Here is a clear and concise description of what the problem is:

When you use multiple profiles in a single Kodi installation Kodi doesn't load skin set in your profile and loads default Estuary instead.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Kodi should load and use the skin which you set in the profile.

## Actual Behavior

Kodi loads Estuary after logging in.

### To Reproduce

1. Create a second profile in Kodi
2. Set **Show login screen** on startup to ON
3. Log in to your second profile, set your skin to anything else than Estuary
4. Restart Kodi, log in to your second profile again

### Debuglog

The debuglog can be found here:
[https://paste.kodi.tv/puhapometa.kodi](url)

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [X] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
Kodi 18 latest nightly - ace36f69

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by pkerling** [comment]
Hello,

thank you for taking time to report this. Unfortunately, the profile functionality is implemented in a fundamentally broken way and is very likely to be removed in Kodi v19. As such, I cannot offer you much hope that this will be fixed in v18. Can you comment on whether this is a regression, i.e. did it work in v17?

**Comment by da-anda** [comment]
note: while it likely will be removed in v19, there are ofc plans to bring it back in newer and a future proof architecture. But we need volunteers for this.

**Comment by Angeloras** [comment]
Hi,

yeah, it did work in Kodi 17 without problems.

My heart almost stopped when I read it will be removed, but I'll keep my fingers crossed for it to return someday. I know not many people are using this feature but it's super useful for my family device with 4 separate profiles. I know you can use the portable mode on Windows to have 4 different Kodi folders but it's a no go on other platforms.

**Comment by pkerling** [comment]
> I know you can use the portable mode on Windows to have 4 different Kodi folders but it's a no go on other platforms.

FWIW, you should be able to do this without much effort on Linux, too.

**Comment by pkerling** [comment]
> yeah, it did work in Kodi 17 without problems.

If you want to go through the effort of bisecting to find out which commit changed the behavior, it *might* be a simple fix - cannot make any promises though :-(

**Comment by garbear** [comment]
Unfortunately this is unlikely to help, because even if you find the commit it was probably made for fundamental architectural changes and can't be reverted. Still worth a try though.

**Comment by Lanternon** [comment]
Yeah, I encountered this bug on all three platforms - Android (Shield TV), Windows and Linux (LibreELEC).
I'm more than willing to find out which commit did this so I'll try older builds (if I remember correctly it lasts for many months - my guess is since April/May 2018) and I'll let you know.
Thank you so much for you time guys.

**Comment by DaVukovic** [comment]
That's a bit strange as I can't reproduce it on Linux (Ubuntu) and Kodi compiled from master. I have 2 profiles at this instance. One is using Confluence as the skin and the other profile is using Aeon Nox. I can easily switch the profiles and each profile loads the specific skin. 

Will upload a logfile later which shows what I'm doing

**Comment by Lanternon** [comment]
> 
> 
> That's a bit strange as I can't reproduce it on Linux (Ubuntu) and Kodi compiled from master. I have 2 profiles at this instance. One is using Confluence as the skin and the other profile is using Aeon Nox. I can easily switch the profiles and each profile loads the specific skin.
> 
> Will upload a logfile later which shows what I'm doing

Well...that's new for me. I always exit Kodi before switching between profiles on Android and turn off my LibreELEC machine. But if you log off it behaves like this for me:
1. You have 2 profiles - A and B.
2. When you start Kodi and login directly to profile B, it loads skin from profile A but not completely - you can't see widgets in Estuary, in Confluence you can't see labels in the main menu...
2a. When you start Kodi and login directly to profile A, it loads skin from profile A but again not completely
3. When you log off and login back to profile B, it loads the correct skin from profile B

So only the first login loads the wrong skin. It's strange but it will definitely help me because logging off would be a backup solution for me.

**Comment by DaVukovic** [comment]
> > That's a bit strange as I can't reproduce it on Linux (Ubuntu) and Kodi compiled from master. I have 2 profiles at this instance. One is using Confluence as the skin and the other profile is using Aeon Nox. I can easily switch the profiles and each profile loads the specific skin.
> > Will upload a logfile later which shows what I'm doing
> 
> Well...that's new for me. I always exit Kodi before switching between profiles on Android and turn off my LibreELEC machine. But if you log off it behaves like this for me:
> 
>     1. You have 2 profiles - A and B.
> 
>     2. When you start Kodi and login directly to profile B, it loads skin from profile A but not completely - you can't see widgets in Estuary, in Confluence you can't see labels in the main menu...
>        2a. When you start Kodi and login directly to profile A, it loads skin from profile A but again not completely
> 
>     3. When you log off and login back to profile B, it loads the correct skin from profile B
> 
> 
> So only the first login loads the wrong skin. It's strange but it will definitely help me because logging off would be a backup solution for me.

Thanks for mentioning the steps again. And you are right. I can reproduce that on Linux as well :+1:

btw, agree with @Razzeee sometimes we forget to thank for those reports. ;)

**Comment by DaVukovic** [comment]
@Lanternon 

I know, it's not a fix, but it works for me after a few tests. You could use a autoexec.py and fill it with:

```
import xmbc
xmbc.executebuiltin('ReloadSkin()')
```

That will load the specific skin with all textures, labels and such. I couldn't find any issue after using that workaround. 

Maybe the helps for the time being.

**Comment by lrusak** [comment]
Profiles will most likely be removed before they can be improved. You can only patch so much. It's better to clean up and start from scratch with a better implementation.

Sorry that the free software isn't good enough for you. You can have your money back 😁

**Comment by lukjak** [comment]
Ekhm, I will keep a bright reply of "right, I got quality and support I paid for" for another occasion.
At least good to have early warning...

**Comment by da-anda** [comment]
issue is, the profile support is hacked into so many sub components of KODI and a change on one end causes issues on the other end. It became more noticeable now that we actually try to cleanup the core and untangle all the interwoven mess. The main focus now relies on untangling the main core components and fine a proper way to resolve "dependencies" during startup, and profiles almost all the time cause some issues during this process. So our idea is to rip them out, get the core into shape and then re-implement them in a much better and more flexible way (like syncing watched states etc).

So yeah, it sucks that we have to rip them out, I also use profiles, but TBH, using them atm also sucks and I rather not have profiles for one version than having issues with them for the rest of my life

**Comment by DaVukovic** [comment]
@lukjak 

For the time being you could try the workaroung I posted above. This is known working and was also tested at forums.

**Comment by Lanternon** [comment]
My final solution for this problem was setting up Autoexec.py in the main userdata folder with this:

import xbmc
xbmc.executebuiltin('System.LogOff()')

and setting up Kodi to login to main master profile after starting up. So after launching Kodi, it logs in and logs out very fast (it's so fast on my Nvidia Shield that you can't see it) and then you can log to any profile you want. It's the second login so the skin loads normally.

Ta-dah! :)

**Comment by DaVukovic** [comment]
@Lanternon 

see: https://github.com/xbmc/xbmc/issues/14732#issuecomment-435525995

;)

**Comment by Lanternon** [comment]
@DaVukovic 

Yeah, I know, but this solution doesn't look good to my eyes, because it takes 1-5 seconds before it loads the skin and you can see Estuary after logging in.
It was my first workaround but then I changed it to logging on -> off automatically because it feels more "out of eyes, out of mind". :)

**Comment by in10se01** [comment]
@Lanternon

Thanks for your effort and sharing the workaround. How do I setup kodi to always automatically boot and login into master user?

Is it somewhere in the settings? I am using a special skin and I don't see an option to chose this. Is it only seen in default kodi skin?

Or is it an additional addon?

Would be happy about a guide here.

Thank you
